### PR TITLE
Implement Velocity VarInt optimizations

### PR DESCRIPTION
**Pull request by Gerrygames**
https://github.com/PaperMC/Paper/pull/6957

### CODE REVIEW

File: patches/server/0919-Optimize-VarInts.patch [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
++++ b/src/main/java/net/minecraft/network/FriendlyByteBuf.java
+@@ -87,15 +87,18 @@ public class FriendlyByteBuf extends ByteBuf {
+         this.source = parent;
+     }
+ 
+-    public static int getVarIntSize(int value) {
+-        for (int j = 1; j < 5; ++j) {
+-            if ((value & -1 << j * 7) == 0) {
+-                return j;
+-            }
```
Comment (Owen1212055) [inline patches/server/0919-Optimize-VarInts.patch]: Don't remove this method, it's making the diff kinda nasty.

File: patches/server/0919-Optimize-VarInts.patch [hunk 2]
```diff
@@ -0,0 +1,59 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: astei <andrew@steinborn.me>
+Date: Sat, 1 Oct 2022 09:56:47 +0200
+Subject: [PATCH] Optimize VarInts
+
+
```
Comment (Owen1212055) [inline patches/server/0919-Optimize-VarInts.patch]: Please link the optimization 
https://github.com/PaperMC/Paper/pull/6957#issuecomment-985250854

would be nice to have some context what this is doing and why

File: patches/server/0919-Optimize-VarInts.patch [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+     }
++    //Paper end - Optimize VarInts
+ 
+     public static int getVarLongSize(long value) {
+         for (int j = 1; j < 10; ++j) {
+@@ -503,7 +506,22 @@ public class FriendlyByteBuf extends ByteBuf {
+         return new UUID(this.readLong(), this.readLong());
+     }
+ 
++    // Paper start - Optimize VarInts
```
Comment (Owen1212055) [inline patches/server/0919-Optimize-VarInts.patch]: move paper comment into the method

**Review by Owen1212055 — CHANGES_REQUESTED** [review]

File: patches/server/0948-Optimize-VarInts.patch [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ 
+     public static int getVarIntSize(int value) {
++    //Paper start - Optimize VarInts
++        return VARINT_EXACT_BYTE_LENGTHS[Integer.numberOfLeadingZeros(value)]; // Paper - Optimize VarInts
++    }
++    private static final int[] VARINT_EXACT_BYTE_LENGTHS = new int[33];
++    static {
++        for (int i = 0; i <= 32; ++i) {
++            VARINT_EXACT_BYTE_LENGTHS[i] = (int) Math.ceil((31d - (i - 1)) / 7d);
++            VARINT_EXACT_BYTE_LENGTHS[32] = 1; // Special case for the number 0.
```
Comment (Owen1212055) [inline patches/server/0948-Optimize-VarInts.patch]: Is there any reason why this is inside the loop? (The special case)

**Review by Owen1212055 — APPROVED** [review]
Tested and everything _seems_ fine but would appreciate it if someone more knowledgeable in this area could take a look.

File: patches/server/0948-Optimize-VarInts.patch (hunk 4 above)
Comment (Gerrygames) [reply patches/server/0948-Optimize-VarInts.patch]: No. Looks like a copy pasta problem. Will fix soon.

**Review by electronicboy — APPROVED** [review]

**Pull request merged**